### PR TITLE
Demo deletion

### LINKED ISSUE #1922: Implement demo deletion
An endpoint for demo deletion is required. It should also be supported on frontend

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR closes #1922

### What's done:
 * Implemented demo deletion
 * Added frontend button that would delete demo
 * Added `ttlSecondsAfterFinished` to `s3-deployment.yaml` in order to clean the startup job (`minio-create-bucket`)

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt [hunk 1]
```diff
@@ -333,4 +333,44 @@ class DemoManagerController(
                     )
                 )
         }
+
+    @PostMapping("/{organizationName}/{projectName}/delete")
+    @RequiresAuthorizationSourceHeader
+    @PreAuthorize("hasRole('ROLE_SUPER_ADMIN')")
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt:339]: do we still need it taking into account that you check permission on project level?

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/service/DemoService.kt [hunk 2]
```diff
@@ -104,4 +107,17 @@ class DemoService(
...
+    /**
+     * @param demo [Demo] entity
+     * @param version version of demo
+     * @return [Mono] of [Unit]
+     */
+    fun delete(demo: Demo, version: String): Mono<StringResponse> = stop(demo)
+        .let { dependencyStorage.list(demo, version) }
+        .concatMap { dependencyStorage.delete(it) }
+        .collectList()
+        .publishOn(Schedulers.boundedElastic())
```
Comment (nulls) [inline save-demo/src/main/kotlin/com/saveourtool/save/demo/service/DemoService.kt:120]: why not `blockingToMono`?

Comment (sanyavertolet) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/service/DemoService.kt:120]: Well, it is actually the same but without additional `Mono` creation

Comment (nulls) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/service/DemoService.kt:120]: yeap, but it's more visual to show blocking operations. but up to you

**Pull request merged**